File: homework_5/randompi.c
// import
#include <stdio.h>
#include <stdlib.h>

// define
#define sqr2(x) ((x)*(x))
#define frand() ((double) rand() / (RAND_MAX))
#define MAXLEN 1000000000

// code
int circumscribed(int radius){
    double xcoord = frand();
    double ycoord = frand();
    double coord = sqr2(xcoord) + sqr2(ycoord);
    
    if(coord <= radius)
        return 1;
    return -1;
}

int main()
{
    int i;
    double circles = 0, rect = 0;;
    for(i = 0; i < MAXLEN; i++)
    {
        if(circumscribed(1) > 0)   		/* got code outline from https://stackoverflow.com/questions/17659652/calculating-pi-using-monte-carlo-method-gives-imprecise-answer */
            circles++;
        rect++;
    }
    
    double PI = (double)4 * circles / rect;
    printf("PI is %2.4f: \n", PI);
    return 0;
}

// end



// recieved help from Aria Druker

File: homework_5/randompi.py
#import
import random

#code
random.seed(42)
total=1000000*1000
out = 0
for i in range(total):
    x = random.random()
    y = random.random()
    if x*x+y*y > 1:
        out = out + 1
print( 4.0*(1.0-float(out)/float(total)))



#got help from Natalie Silva